### PR TITLE
Add frozen string literal

### PR DESCRIPTION
**Pull request by lucianosousa**
Improving app speed with frozen string literal in ruby files

### CODE REVIEW

**Comment by segiddins** [comment]
How much does this improve speed by?

**Comment by lucianosousa** [comment]
@segiddins 

master branch:
```
$ time bundle exec rake
real 2m20.122s
user 1m19.148s
sys  0m11.428s
```

this branch:
```
$ time bundle exec rake
real 2m15.875s
user 1m19.849s
sys  0m11.186s
```

Despite the performance is not a huge advantage, I would say that a second reason why we should merge this PR is the same described here: https://www.pluralsight.com/blog/software-development/ruby-2-3--working-with-immutable-strings-
For consistency/ easy maintenance, it's good keep strings immutable

File: app/controllers/api/compact_index_controller.rb [hunk 1]
```diff
@@ -30,7 +32,7 @@ def set_compact_index_cache_headers
   end
 
   def render_range(response_body)
-    headers['ETag'] = '"' << Digest::MD5.hexdigest(response_body) << '"'
+    headers['ETag'] = +'"' << Digest::MD5.hexdigest(response_body) << '"'
```
Comment (sonalkr132) [inline app/controllers/api/compact_index_controller.rb]: please use `.dup`, `+` doesn't seem obvious (I don't even know about this operator :sweat_smile: ).

File: app/models/version.rb [hunk 2]
```diff
@@ -301,7 +303,7 @@ def to_gem_version
   end
 
   def to_install
-    command = "gem install #{rubygem.name}"
+    command = +"gem install #{rubygem.name}"
```
Comment (sonalkr132) [inline app/models/version.rb]: Are you sure `+` is need here? use `.dup`, if it is.

**Comment by lucianosousa** [comment]
@sonalkr132 updated

File: app/controllers/api/compact_index_controller.rb [hunk 3]
```diff
@@ -30,7 +32,7 @@ def set_compact_index_cache_headers
   end
 
   def render_range(response_body)
-    headers['ETag'] = '"' << Digest::MD5.hexdigest(response_body) << '"'
+    headers['ETag'] = '"' << Digest::MD5.hexdigest(response_body) << '"'.dup
```
Comment (sonalkr132) [inline app/controllers/api/compact_index_controller.rb]: Apparently unary plus is faster :sweat: Can you please change it back to `+`? Sorry.
Performance/UnfreezeString: Use unary plus to get an unfrozen string literal.
https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Performance/UnfreezeString

Comment (lucianosousa) [reply app/controllers/api/compact_index_controller.rb]: yeah man! ;p I've seen this before, but I also agree that the syntax is not so fine at all

**Comment by olliebennett** [comment]
You might find some unexpected behaviours/bugs arising after you configure all strings to be frozen (as you're doing here) - I did.

If test coverage is good, you should get away without any problems, although a more measured approach might be to work through an exclusion list in `rubocop_todo.yml` to tackle one subset of the codebase at a time. This allows smaller more iterative deploys and you can test and tackle any compatibility problems. The pleasant side-effect is to reduce the size / surface area of this PR.

For example, in `rubocop_todo.yml`;

```yaml
Style/FrozenStringLiteralComment:
  Exclude:
    - 'app/controllers/**/*.rb'
    - 'app/helpers/**/*.rb'
    - 'app/jobs/**/*.rb'
    - 'app/mailers/**/*.rb'
    - 'app/models/**/*.rb'
    - 'config/initializers/**/*.rb'
    - 'test/**/*.rb'
```

**Pull request closed without merging**